app/services/concerns/xccdf/rule_group_rules.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving RuleReferencesRules
+  module RuleGroupRules
+    extend ActiveSupport::Concern
+
+    included do
+      def save_rule_group_rules
+        # Setting up the 4 different associations for requiring/conflicting rule groups or rules
+        required_for_rules
+        required_for_rule_groups
+        conflicted_for_rules
+        conflicted_for_rule_groups
+      end
+
+      private
+      # These next 4 methods can probably done in a more efficient/cleaner way but just wanted to get them started
+      def required_for_rules
+        @op_rules.each do |op_rule|
+          op_rule.requires.each do |required_ref_id|
+            if rule_for(ref_id: required_ref_id)
+              rule_for(ref_id: op_rule.id).required_rules << rule_for(ref_id: required_ref_id)
```

Comment:
AFAIK these `<<` aren't offloaded, so this would be extremely slow. I think you should use [import!](https://github.com/zdennis/activerecord-import) as we do it with other things that we import. There's some inspiration in the PR I sent you to review :wink:

Reply:
I see what you mean with the import! part but I'm not sure how to add the objects to the associations without the `<<`.. I was reading through [this](https://api.rubyonrails.org/v7.0.2/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) and thinking I could maybe use `.build` but it seemed like << was less confusing since there are so many different associations
I also was thinking that I could create an array of objects that need to be added onto an association and then push them all on at once instead of looping and pushing them on one by one. Maybe that'd help with efficiency? Or do you think build might be better?

Reply:
You don't need the `Rule` model for this at all, you can just take its ID and pair it in a `RuleGroupRule` object. Build an array of these objects and call the import on them. Check out how `Xccdf::ProfileRules` imports stuff, something similar.

Reply:
Okay, that's what I was thinking but if we're taking a `Rule` or `RuleGroup` and pairing it in a RuleGroupRule, how will we know if it's required or conflicting? Like if we add it to the joins table, how does it know which association it belongs to? Since there are multiple

Reply:
Okay, checking 😄